ConfigurationSystem/Agent/RucioSynchronizer.py

```diff
@@ -0,0 +1,359 @@
+
+        read_protocols = {}
+        protocols = se.options.get('AccessProtocols')
+        if not protocols:
+            continue
+        idx = 1
+        for prot in protocols:
+            read_protocols[prot] = idx
+            idx +=1
+            if prot not in all_protocols:
+                all_protocols.append(prot)
+        write_protocols = {}
+        protocols = se.options.get('WriteProtocols')
+        if not protocols:
+            continue
+        idx = 1
+        for prot in protocols:
+            write_protocols[prot] = idx
+            idx +=1
+            if prot not in all_protocols:
+                all_protocols.append(prot)
+
+
+        mapping = {'Protocol': 'scheme', 'Host': 'hostname', 'Port': 'port', 'Path': 'prefix'}
+        for protocol in all_protocols:
+            space_token = None
+            params = {'hostname': None,
+                      'scheme': None,
+                      'port': None,
+                      'prefix': None,
+                      'impl': 'rucio.rse.protocols.gfal.Default',
+                      'domains': {"lan": {"read": 0,
+                                          "write": 0,
+                                          "delete": 0},
+                                  "wan": {"read": 0,
+                                          "write": 0,
+                                          "delete": 0,
+                                          "third_party_copy": 0}}}
+            res = se.getStorageParameters(protocol=protocol)
+            if res['OK']:
+                values = res['Value']
+                for key in ['Protocol', 'Host', 'Access', 'Path', 'Port', 'WSUrl', 'SpaceToken', 'WSUrl', 'PluginName']:
+                    value = values.get(key)
+                    if key in mapping:
+                        params[mapping[key]] = value
+                    else:
+                        if key == 'SpaceToken':
+                            space_token = value
+                        if params['scheme'] == 'srm' and key == 'WSUrl':
+                            params['extended_attributes'] = {'web_service_path': '%s' % value, 'space_token': space_token}
+                    if key == 'Protocol':
+                        params['domains']['lan']['read'] = read_protocols.get(value, 0)
+                        params['domains']['wan']['read'] = read_protocols.get(value, 0)
+                        params['domains']['lan']['write'] = write_protocols.get(value, 0)
+                        params['domains']['wan']['write'] = write_protocols.get(value, 0)
+                        params['domains']['lan']['delete'] = write_protocols.get(value, 0)
+                        params['domains']['wan']['delete'] = write_protocols.get(value, 0)
+                        params['domains']['wan']['third_party_copy'] = write_protocols.get(value, 0)
+                seProtocols[seName].append(params)
+    return S_OK(seProtocols)
+
+
+class RucioSynchronizer(AgentModule):
```

Comment:
We always names the agents with "blaBlaAgent", and with the notable exception of the SiteDirector, this is a convention everywhere. So, this class, and the file name, should be "RucioSynchronizerAgent".